internal/controllers/config_manager.go

```diff
@@ -143,14 +144,15 @@ func (c *KubeEnvoyConfigManager) UpdateConfiguration(ctx context.Context, fleetI
 			return fmt.Errorf("failed to generate options from the static route config: %w", err)
 		}
 
-		if err := UpdateConfigFromOpts(envoyConfig, opts); err != nil {
-			return fmt.Errorf("failed to generate config: %w", err)
+		kubernetesClient := c.Client
```

Comment:
why make a copy here in the first place when it can be passed to the function as is

Reply:
Resolved.